tests/util.py

```diff
@@ -51,8 +55,18 @@ def download_file(url: str, filename: Path | str) -> str:
 
     url = convert_google_drive_link(url)
 
-    path, _ = urlretrieve(url, filename=filename)
-    return path
+    temp_filename = filename.with_suffix(f".part-{int(time.time())}")
+
+    try:
+        logger.info("Downloading %s to %s", url, filename)
+        path, _ = urlretrieve(url, filename=temp_filename)
+        temp_filename.rename(filename)
+        return path
```

Comment:
This is wrong. Since you are downloading to a temp file first, `path` will be that temp file. You need to return a string version of `filename`.

Honestly, I have no idea what we were thinking when we made this function return what is essentially `filename`. I think the best fix would be to make this function not return anything at all. You just need to account for this change in `from_url_zip` and then everything should be good.

Reply:
Yeah, I don't know what I was thinking either 😁 Rebased, added a commit that removes the retval 👍